builder/internals.go

```diff
@@ -128,6 +128,8 @@ func (b *Builder) commit(id string, autoCmd *runconfig.Command, comment string)
 	if err != nil {
 		return err
 	}
+	b.Daemon.Graph().Retain(b.id, image.ID)
+	b.activeImages = append(b.activeImages, image.ID)
 	b.image = image.ID
```

Comment:
There's still a race condition: the time between the `b.Daemon.Commit()` call above and the call to `b.Daemon.Graph().Retain()` here, though I guess there's really not much we can do about that.